test/ConfigDistributor.spec.js

```diff
@@ -1,198 +1,121 @@
-
-    context('when supplied with a JavaScript object defining data for the JavaScript layer', () => {
-
-      let jsConfigFixture;
-
-      beforeEach(() => {
-        jsConfigFixture = fixtures.jsConfigToProcess.input;
-      });
+  it('initiates config consolidation with the supplied paths', () => {
+    const consolidatorSpy = sinon.spy(consolidatorFixtures.forJsAndSass, 'consolidate');
+    const expectedPaths = paths.config;
 
-      it('correctly converts the JavaScript into the appropriate JSON', () => {
-        const processed = ConfigDistributor.processForJs(['breakpoints'], jsConfigFixture);
-        expect(processed).to.deep.equal(JSON.stringify(fixtures.jsConfigToProcess.input));
+    return distributor.distribute(consolidatorFixtures.forJsAndSass)
+      .then(() => {
+        return expect(consolidatorSpy.calledOnceWithExactly(expectedPaths)).to.be.true;
       });
-
-    });
-
   });
 
-  describe('writeDirectory static method', () => {
+  context('when deriving the data to distribute to the respective layers', () => {
 
-    context('when there is an error creating the directory', () => {
+    it('determines the correct data to distribute to the JavaScript layer', () => {
+      const expectedData = JSON.stringify(cannedData.expectedOutput.js);
 
-      it('returns a promise that will be rejected', () => {
-        const invalidDirName = (function () {
-          const parts = new Array(256);
-          parts.fill('a');
-          return parts.join('');
-        }());
-        return expect(
-          ConfigDistributor.writeDirectory(invalidDirName)
-        ).to.be.rejected;
-      });
+      filesystemMock.expects('writeFile').once().withArgs(expectedData);
 
+      return distributor.distribute(consolidatorFixtures.forJsOnly)
+        .then(() => {
+          filesystemMock.verify();
+        });
     });
 
-    context('when there no error', () => {
-
-      let validDirName;
+    it('determines the correct data to distribute to the Sass layer', () => {
+      const expectedData = cannedData.expectedOutput.sass;
 
-      beforeEach(() => {
-        validDirName = path.join(__dirname, '/tmp/');
-        if (fs.existsSync(validDirName)) {
-          fs.rmdirSync(validDirName);
-        }
-        expect(fs.existsSync(validDirName)).to.be.false;
-      });
+      filesystemMock.expects('writeFile').once().withArgs(expectedData);
 
-      afterEach(() => {
-        fs.rmdirSync(validDirName);
-      });
-
-      it('returns a promise that will not be rejected', () => {
-        return expect(
-          ConfigDistributor.writeDirectory(validDirName)
-        ).to.not.be.rejected;
-      });
-
-      it('writes the directory', () => {
-        return ConfigDistributor.writeDirectory(validDirName).then(() => {
-          return expect(fs.existsSync(validDirName)).to.be.true;
+      return distributor.distribute(consolidatorFixtures.forSassOnly)
+        .then(() => {
+          filesystemMock.verify();
```

Comment:
I wonder whether this could be pushed in `tearDown`. It's a groundbreaking change so leave it for some other time.